### PR TITLE
pbio/drv/imu_lsm6ds3tr_c_stm32: Read gyro and acceleration in one pass.

### PR DESCRIPTION
**Pull request by laurensvalk**
This should further reduce overhead.

### CODE REVIEW

**Comment by laurensvalk** [comment]
When we do this, we technically shouldn't even need a process anymore.

We could just start reading on the pin1 interrupt, and do the memcpy on getting the read_complete interrupt.

I suppose error handing is easier with the process, let's keep an eye on whether the reset is still triggered a lot.

File: lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c [hunk 1]
```diff
@@ -213,6 +213,9 @@ static PT_THREAD(pbdrv_imu_lsm6ds3tr_c_stm32_init(struct pt *pt)) {
         .int1_drdy_g = 1,
     }));
 
+    // Enable rounding mode so we can get gyro + accel in one read.
+    PT_SPAWN(pt, &child, lsm6ds3tr_c_rounding_mode_set(&child, ctx, LSM6DS3TR_C_ROUND_GY_XL));
```
Comment (dlech) [inline lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c:217]: This is not needed. We only read 12 bytes so we never hit the rounding point.

File: lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c [hunk 2]
```diff
@@ -228,7 +231,7 @@ PROCESS_THREAD(pbdrv_imu_lsm6ds3tr_c_stm32_process, ev, data) {
     I2C_HandleTypeDef *hi2c = &imu_dev->hi2c;
 
     static struct pt child;
-    static uint8_t buf[6];
+    static uint8_t buf[12];
```
Comment (dlech) [inline lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c:234]: We are using 12 enough times we should probably introduce a `#define` for it.

File: lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c (hunk 1 above)
Comment (dlech) [reply lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c:217]: Alternately to actually take advantage of this, we would need to use different I2C functions to write the register once outside of the loop and then read inside of the loop (`HAL_I2C_Master_Transmit_IT` and `HAL_I2C_Master_Receive_IT` or possibly `HAL_I2C_Master_Sequential_Receive_IT`?) and a goto after error recovery to write the register again.

Comment (laurensvalk) [reply lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c:217]: Or something like the note above with no loop at all?

Comment (dlech) [reply lib/pbio/drv/imu/imu_lsm6ds3tr_c_stm32.c:217]: Doing more in the interrupt handlers sounds dangerous to me. For example, if we call HAL functions from there, they may have (short) blocking busy loop waits that we don't know about. The less we do in interrupt handlers the better.

And I don't see how that would make using the rounding function work either.

**Comment by dlech** [comment]
I went ahead and made the suggested changed and fixed some other unrelated issues.

**Pull request merged**